functions/src/modules/SQR.functions.ts

```diff
@@ -125,111 +125,130 @@ export const sendEmailOnNewAllotment = functions.database
 
-export const processSubmissions = functions.database
-  .ref('/webforms/sqr/{submission_id}')
-  .onCreate(async (snapshot, context) => {
-    const original = snapshot.val();
-
-    let audioFileStatus = 'WIP';
-
-    if (original.not_preferred_language) audioFileStatus = 'Spare';
-    else if (original.unable_to_play_or_download)
-      audioFileStatus = 'Audio Problem';
-
-    // 1. Add the webform data to a SQR submissions DB path
-    let submission = {
-      fileName: original.audio_file_name,
-      cancellation: {
-        notPreferredLanguage: audioFileStatus === 'Spare',
-        audioProblem: audioFileStatus === 'Audio Problem',
-      },
-      soundQualityRating: original.sound_quality_rating,
-      unwantedParts: original.unwanted_parts,
-      soundIssues: original.sound_issues,
-      duration: {
-        beginning: original.beginning,
-        ending: original.ending,
-      },
-      comments: original.comments,
-      token: original.token,
-      created: original.created,
-      //  timestamp of the submission creation,
-      // can differ from COMPLETED in case of saving a DRAFT and completing later.
-      completed: original.completed, // timestamp of the submission completion.
-      changed: original.changed, //timestamp of the submission update.
-      devotee: {
-        name: original.name,
-        emailAddress: original.email_address,
-      },
-    };
-    db.ref(`/sqr/submissions/${original.serial}`).update(submission);
-
-    // 2. Update the allotment ( first get the previous NOTES )
-    const filesSnapshot = await db
-      .ref(`/sqr/files/${original.list}/${original.audio_file_name}`)
-      .once('value');
-    let allotmentUpdates = { status: audioFileStatus };
-    // in case 1 & 2 add the comments to the notes
-    if (audioFileStatus !== 'WIP')
-      allotmentUpdates['notes'] = `${filesSnapshot.val().notes}\n${
-        original.comments
-      }`;
-    // if the audio has a problem then REMOVE the devotee from the file allotment
-    if (audioFileStatus === 'audioProblem') allotmentUpdates['devotee'] = {};
-    db.ref(`/sqr/files/${original.list}/${original.audio_file_name}`).update(
-      allotmentUpdates
-    );
-
-    // Coordinator object example { templateid: 3, email:'a@a.a', name: 'Aj' }
-    let coordinator = functions.config().sqr.coordinator;
-    let templateId = functions.config().sqr.allotment.templateid;
-    // 3. Notify the coordinator
-    // 3.1 Get the submitted audio file data
+let prepareSubmissionParams = (original, audioFileStatus) => {
+  if (original.not_preferred_language) audioFileStatus = 'Spare';
+  else if (original.unable_to_play_or_download)
+    audioFileStatus = 'Audio Problem';
+
+  // 1. Add the webform data to a SQR submissions DB path
+  let submission = {
+    fileName: original.audio_file_name,
+    cancellation: {
+      notPreferredLanguage: audioFileStatus === 'Spare',
+      audioProblem: audioFileStatus === 'Audio Problem',
```

Comment:
In the above two lines it is better not to rely on the `audioFileStatus`, but rather on the original properties. Then you will not need this `audioFileStatus` in this function at all as I pointed out in my comment above.

Reply:
Sorry I haven't made any new change to this submission Object i just tried to fit it into a function. The code is the same as the previous one.

Reply:
I give you permission to change the code :)

`notPreferredLanguage` should be taken from `not_preferred_language`, and
`audioProblem` should be taken from `unable_to_play_or_download`.